pkg/kubelet/server.go

```diff
@@ -346,6 +352,18 @@ func (s *Server) InstallDebuggingHandlers() {
 	s.restfulCont.Add(ws)
 }
 
+// InstallApiHandlers registers the HTTP request patterns that serve the kubelet versioned api.
+func (s *Server) InstallApiHandlers() {
+	ws := new(restful.WebService).
+		Path("/api/v1/").
+		Consumes(restful.MIME_JSON).
+		Produces(restful.MIME_JSON)
+
+	ws.Route(ws.GET("rawMetrics").To(handlers.RawMetricsHandler(s.host, s.host.GetStatusManager())))
```

Comment:
`rawNodeMetrics` should *not* include pod Metrics.

On Thu, Oct 15, 2015 at 12:18 PM, Tim St. Clair <notifications@github.com>
wrote:

> In pkg/kubelet/server.go
> <https://github.com/kubernetes/kubernetes/pull/15691#discussion_r42167748>
> :
>
> > @@ -346,6 +352,18 @@ func (s *Server) InstallDebuggingHandlers() {
> >  	s.restfulCont.Add(ws)
> >  }
> >
> > +// InstallApiHandlers registers the HTTP request patterns that serve the kubelet versioned api.
> > +func (s *Server) InstallApiHandlers() {
> > +	ws := new(restful.WebService).
> > +		Path("/api/v1/").
> > +		Consumes(restful.MIME_JSON).
> > +		Produces(restful.MIME_JSON)
> > +
> > +	ws.Route(ws.GET("rawMetrics").To(handlers.RawMetricsHandler(s.host, s.host.GetStatusManager())))
>
> In this model, would the rawNodeMetrics still include pod metrics, or
> would it just be machine + system containers?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15691/files#r42167748>.
>